### PR TITLE
June 2024 Upstream Merge

### PR DESCRIPTION
**Pull request by whatston3**
## About the PR

Merges against space-wizards' space-station-14 after #29100 (updating RobustToolbox to 226.0.0).

## Why / Balance

It is time.

## How to test

Very carefully.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** ***this PR does not require an ingame showcase***

## Breaking changes

Everything.

**Changelog**

Nope.

### CODE REVIEW

File: Content.Client/Preferences/UI/HumanoidProfileEditor.xaml.cs [hunk 1]
```diff
@@ -1238,3 +1239,4 @@ private void OnCheckBoxToggled(BaseButton.ButtonToggledEventArgs args)
         }
     }
 }
+*/
```
Comment (dvir001) [inline Content.Client/Preferences/UI/HumanoidProfileEditor.xaml.cs:1242]: Confused screaming

File: Content.Client/Preferences/UI/HumanoidProfileEditor.Random.cs [hunk 2]
```diff
@@ -21,3 +22,4 @@ private void RandomizeName()
         }
     }
 }
+*/
```
Comment (dvir001) [inline Content.Client/Preferences/UI/HumanoidProfileEditor.Random.cs:25]: Confused screaming

File: Content.Client/Preferences/UI/CharacterSetupGui.xaml.cs [hunk 3]
```diff
@@ -274,3 +275,4 @@ protected override void Dispose(bool disposing)
         }
     }
 }
+*/
```
Comment (dvir001) [inline Content.Client/Preferences/UI/CharacterSetupGui.xaml.cs:278]: Confused screaming

File: Resources/Prototypes/_NF/Roles/Jobs/Hostile/syndicate_naval_forces.yml [hunk 4]
```diff
@@ -7,9 +7,10 @@
     shoes: ClothingShoesBootsCombat
     gloves: ClothingHandsGlovesColorBlackPython
     neck: ClothingNeckScarfStripedSyndieRed
-    mask: ClothingMaskGasSyndicate
+#    mask: ClothingMaskGasSyndicate
```
Comment (dvir001) [inline Resources/Prototypes/_NF/Roles/Jobs/Hostile/syndicate_naval_forces.yml:10]: Need to restore all masks later

**Comment by dvir001** [comment]
Reminder I need to whitelist new emotes on some of the races or they wont work.

File: Resources/Prototypes/_NF/Events/events.yml
[the remark this answers is not in the document]
Comment (whatston3) [reply Resources/Prototypes/_NF/Events/events.yml:1]: Took another look at this.

What exactly needs to be done?  Should the commented out events be removed?  As far as I can tell, the only thing that's changed with the uncommented events is the delay between announcement and the start of the event (e.g. BreakerFlip, PowerGridCheck take longer to start than usual).  FalseAlarm was commented out upstream, but I'll return it just for the sake of consistency before the merge.

I've taken account of the events that existed on the head of frontier's master before this PR and space-station-14's head that we're merging in.  Once FalseAlarm is restored, the difference is listed below.

**Events that exist on space-station-14's head but not frontier's head:**
Less dangerous:
* BluespaceArtifact
* BluespaceLocker
* BureaucraticError
* ClericalError
* ClosetSkeleton
* KudzuGrowth

More dangerous(?):
* DragonSpawn
* NinjaSpawn
* RevenantSpawn
* KingRatMigration
* SlimesSpawn
* SpiderSpawn
* SpiderClownSpawn
* ZombieOutbreak
* SleeperAgentsRule
* ImmovableRodSpawn
* MimicVendorRule

**Comment by dvir001** [comment]
For the love of god pass tests

**Comment by ErhardSteinhauer** [comment]
Submodule update in pr found (pull_request)?

**Comment by dvir001** [comment]
Merge time

**Review by dvir001 — APPROVED** [review]
Real

image

**Comment by ErhardSteinhauer** [comment]
Oh boy..

**Pull request merged**